CUDADataFormats/EcalRecHitSoA/interface/EcalUncalibratedRecHit_soa.h

```diff
@@ -0,0 +1,74 @@
+#ifndef CUDADataFormats_EcalRecHitSoA_interface_EcalUncalibratedRecHit_soa_h
+#define CUDADataFormats_EcalRecHitSoA_interface_EcalUncalibratedRecHit_soa_h
+
+#include <vector>
+#include <array>
+
+#include "DataFormats/EcalDigi/interface/EcalDataFrame.h"
+
+#include "CUDADataFormats/EcalRecHitSoA/interface/RecoTypes.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/CUDAHostAllocator.h"
+
+namespace ecal {
+
+namespace Tag {
+
+struct soa {};
+struct ptr {};
+
+}
+
+template<typename T, typename L = Tag::soa>
```

Comment:
I admit I'm confused by the `typename L = Tag::soa` template parameter,  because I do not see it used anywhere ?

Reply:
- `Tag::soa` is used for the host side output, and is sitting in the producer itself.
- `Tag::ptr` is used for device side output and can be found in `DeclsForKernels.h`, etc...